ocaml/xapi/xapi_host.ml

```diff
@@ -2671,44 +2671,165 @@ let with_temp_file_contents ~contents f =
     )
     (fun () -> Sys.remove filename)
 
-let write_uefi_certificates_to_disk ~__context ~host:_ =
-  if
-    Sys.file_exists !Xapi_globs.varstore_dir
-    && Sys.is_directory !Xapi_globs.varstore_dir
-  then
-    match
-      Base64.decode
-        (Db.Pool.get_uefi_certificates ~__context
-           ~self:(Helpers.get_pool ~__context)
-        )
-    with
-    | Ok contents ->
-        (* Remove existing certs before extracting xapi ones
-         * to avoid a extract override issue. *)
-        List.iter
-          (fun name ->
-            let path = Filename.concat !Xapi_globs.varstore_dir name in
-            Unixext.unlink_safe path
+let really_write_uefi_certificates_to_disk ~__context ~host:_ ~value =
+  let ( // ) = Filename.concat in
+  match value with
+  | "" ->
+      let files_of_dir path =
```

Comment:
You could use `Sys.readdir` and `Sys.is_directory`. Using more low level functions is important if you want to control the behavior around links.

Reply:
fixed in the last patchset